pkg/component/describe/describe.go

```diff
@@ -234,6 +229,41 @@ func GetRunningOn(ctx context.Context, n string, kubeClient kclient.ClientInterf
 	return runningOn, nil
 }
 
+func filterByPlatform[T platformDependent](ctx context.Context, isFeatEnabled bool, all []T, includeIfFeatDisabled bool) (result []T) {
```

Comment:
Could you please add some unit tests to help understand this logic?

Reply:
Added in [`974d25c` (#6964)](https://github.com/redhat-developer/odo/pull/6964/commits/974d25cf8f06ed0098ff4dbfc1c2e9cedd9ba98e)